device/blackhole/blackhole_coordinate_manager.cpp

```diff
@@ -1,29 +1,115 @@
+            const tt_xy_pair& tensix_core = CoordinateManager::tensix_cores[x + y * grid_size_x];
+            tensix_logical_to_virtual[{x, y}] =
+                CoreCoord(tensix_core.x, tensix_core.y, CoreType::TENSIX, CoordSystem::VIRTUAL);
+            tensix_virtual_to_logical[tensix_core] = CoreCoord(x, y, CoreType::TENSIX, CoordSystem::LOGICAL);
+        }
+    }
+
+    BlackholeCoordinateManager::fill_tensix_logical_to_translated();
+}
+
+void BlackholeCoordinateManager::fill_tensix_logical_to_translated() {
+    const size_t num_harvested_x = __builtin_popcount(CoordinateManager::tensix_harvesting_mask);
+    const size_t grid_size_x = CoordinateManager::tensix_grid_size.x;
+    const size_t grid_size_y = CoordinateManager::tensix_grid_size.y;
+
+    for (size_t x = 0; x < grid_size_x - num_harvested_x; x++) {
+        for (size_t y = 0; y < grid_size_y; y++) {
+            const CoreCoord virtual_coord = CoordinateManager::tensix_logical_to_virtual[{x, y}];
+            const size_t translated_x = virtual_coord.x;
+            const size_t translated_y = virtual_coord.y;
+            CoordinateManager::tensix_logical_to_translated[{x, y}] =
+                CoreCoord(translated_x, translated_y, CoreType::TENSIX, CoordSystem::TRANSLATED);
+            CoordinateManager::tensix_translated_to_logical[{translated_x, translated_y}] =
+                CoreCoord(x, y, CoreType::TENSIX, CoordSystem::LOGICAL);
+        }
+    }
+}
+
+void BlackholeCoordinateManager::dram_harvesting(const size_t dram_harvesting_mask) {
+    CoordinateManager::dram_harvesting_mask = dram_harvesting_mask;
+    CoordinateManager::clear_dram_harvesting_structures();
+
+    size_t num_harvested_banks = __builtin_popcount(dram_harvesting_mask);
+
+    for (size_t x = 0; x < dram_grid_size.x - num_harvested_banks; x++) {
+        for (size_t y = 0; y < dram_grid_size.y; y++) {
+            const tt_xy_pair& dram_core = CoordinateManager::dram_cores[x * dram_grid_size.y + y];
+            CoordinateManager::dram_logical_to_virtual[{x, y}] =
+                CoreCoord(dram_core.x, dram_core.y, CoreType::DRAM, CoordSystem::VIRTUAL);
+            CoordinateManager::dram_virtual_to_logical[dram_core] =
+                CoreCoord(x, y, CoreType::DRAM, CoordSystem::LOGICAL);
+        }
+    }
+
+    size_t logical_x = 0;
+    for (size_t x = 0; x < dram_grid_size.x; x++) {
+        if (!(dram_harvesting_mask & (1 << x))) {
+            for (size_t y = 0; y < dram_grid_size.y; y++) {
+                const tt_xy_pair& dram_core = CoordinateManager::dram_cores[x * dram_grid_size.y + y];
+                CoordinateManager::dram_logical_to_physical[{logical_x, y}] =
+                    CoreCoord(dram_core.x, dram_core.y, CoreType::DRAM, CoordSystem::PHYSICAL);
+                CoordinateManager::dram_physical_to_logical[dram_core] =
+                    CoreCoord(logical_x, y, CoreType::DRAM, CoordSystem::LOGICAL);
+            }
+            logical_x++;
         }
```

Comment:
Left in base coord_manager file, but also here so you don't forget this place as well:
You never fill translated dram coords?

Reply:
Fixed